collect_app/src/main/java/org/odk/collect/android/activities/ScanQRCodeActivity.java

```diff
@@ -0,0 +1,98 @@
+package org.odk.collect.android.activities;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import com.google.zxing.integration.android.IntentIntegrator;
+import com.google.zxing.integration.android.IntentResult;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.listeners.ActionListener;
+import org.odk.collect.android.listeners.PermissionListener;
+import org.odk.collect.android.preferences.AdminSharedPreferences;
+import org.odk.collect.android.preferences.GeneralSharedPreferences;
+import org.odk.collect.android.preferences.PreferenceSaver;
+import org.odk.collect.android.utilities.CompressionUtils;
+import org.odk.collect.android.utilities.LocaleHelper;
+import org.odk.collect.android.utilities.PermissionUtils;
+import org.odk.collect.android.utilities.ToastUtils;
+
+import java.io.IOException;
+import java.util.zip.DataFormatException;
+
+import androidx.annotation.Nullable;
+import timber.log.Timber;
+
+// ScanQRCodeActivity uses the decorator pattern to process the results from ScannerWithFlashlightActivity
```

Comment:
I like the comment but the "decorator" part confused me. Generally I'd think of a decorator adding functionality to an interface. I could totally just be missing something!

Reply:
I guess I was thinking of ScanQRCodeActivity as adding functionality on top of the ScannerWithFlashlightActivity and removing the need to interact with ScannerWithFlashlightActivity directly anymore. Would it be more appropriate that this is the chain of responsibility pattern?

Reply:
I don't think you need to explicitly call out a pattern. I'd change the comment to `Initiates barcode scanning and process its results` or something like that.